### PR TITLE
Pytorch Maskrcnn Scripts Refactor

### PR DESCRIPTION
**Pull request by rondogency**
Please review nightly SM and EC2 scripts PRs together to make it work end-to-end

- Fixed evaluation stage, now evaluation will run to get accuracy if you don't pass skip-test
- Removed train_net_herring.py in favor of train_net.py, which is closer to existing Nvidia scripts
- Also removed train_net_herring_sm.py in herring repo to de-dup scripts
- Made batch_size=4 per GPU default and bypass config files - down the line we just make sure same hp setting doesn't cause regression

EC2 script PR: https://github.com/HerringForks/Herring-Benchmarks/pull/72
SM script PR with full run setting and parser: https://github.com/aws/herring/pull/527

Full run experiment:

Latest Herring:
p4d 8 nodes 6000 steps bs=4
throughput: 902.86 sample/sec
train_time: 1822 sec
BBOX_mAP : 33.42 %
MASK_mAP : 30.51 %

NCCL: (with this PR https://github.com/HerringForks/DeepLearningExamples/pull/50)
p4d 8 nodes 6000 steps bs=4
throughput: 662.35 sample/sec
train_time: 2318 sec
BBOX_mAP : 33.40 %
MASK_mAP : 30.45 %

**We have no difference on accuracy comparing with DDP but 30% faster on end-to-end full training**

sample output from ondemand benchmark:
{
    "bbox_acc":15.923706465453103,
    "rcnn_loss":0.1443,
    "rpn_loss":0.0276,
    "segm_acc":15.001407582320034,
    "throughput":153.84,
    "train_time":847.0
}

### CODE REVIEW

File: PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py [hunk 1]
```diff
@@ -109,21 +115,14 @@ def train(cfg, local_rank, distributed, fp16, dllogger):
...
     if distributed:
-        if use_apex_ddp:
-            model = DDP(model, delay_allreduce=True)
-        else:
-            model = torch.nn.parallel.DistributedDataParallel(
-                model, device_ids=[local_rank], output_device=local_rank,
-                # this should be removed if we update BatchNorm stats
-                broadcast_buffers=False,
-            )
+        model = DDP(model, device_ids=[local_rank], broadcast_buffers=False, bucket_cap_mb=bucket_cap_mb)
```
Comment (yselivonchyk) [inline PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py:118]: Please, leave some comment around `broadcast_buffers`, so we can trace that this behavior is different from default repo

File: PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py [hunk 2]
```diff
@@ -41,12 +47,13 @@
...
-except ImportError:
-    print('Use APEX for better performance')
-    use_apex_ddp = False
+#try:
+#    from apex.parallel import DistributedDataParallel as DDP
+#    use_apex_ddp = True
+#except ImportError:
+#    print('Use APEX for better performance')
+#    use_apex_ddp = False
+use_apex_ddp = False
```
Comment (yselivonchyk) [inline PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py:56]: this comments can be removed

File: PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py [hunk 3]
```diff
@@ -257,7 +286,10 @@ def main():
 
     if args.skip_checkpoint:
         cfg.SAVE_CHECKPOINT = False
-        
+
+    cfg.SOLVER.IMS_PER_BATCH = num_gpus * 4
+    cfg.SOLVER.BASE_LR = num_gpus / 8 * 0.005
```
Comment (yselivonchyk) [inline PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py:291]: Can you please describe what is the reason / effect of larger batch size and where is this LR rule coming from?

**Review by yselivonchyk — APPROVED** [review]

**Review by piyushghai — APPROVED** [review]
